### PR TITLE
updates for teleop

### PR DESCRIPTION
**Pull request by floweisshardt**
superceeds #115 

updated during testing on cob4-2:
- disable light (due to not switching back to default breath mode)
- encapsulate say
- use deadman button to enable switch mode

### CODE REVIEW

File: cob_teleop/ros/src/cob_teleop.cpp [hunk 1]
```diff
@@ -343,14 +345,19 @@ void CobTeleop::updateBase(){
...
 }
 
-
-void CobTeleop::joy_cb(const sensor_msgs::Joy::ConstPtr &joy_msg){
-
-  if(mode_switch_button_>=0 && mode_switch_button_<(int)joy_msg->buttons.size() && joy_msg->buttons[mode_switch_button_]==1)
+void CobTeleop::say(std::string text, bool blocking)
+{
+  cob_sound::SayGoal saygoal;
+  std::replace(text.begin(), text.end(), '_', ' ');
```
Comment (fmessmer) [inline cob_teleop/ros/src/cob_teleop.cpp]: I'd prefer `say` to say what is passed to it.
(Thus I'd prefer to use `replace` where the saytext is composed. Even if that means some duplication)

File: cob_teleop/ros/src/cob_teleop.cpp [hunk 2]
```diff
@@ -382,20 +395,32 @@ void CobTeleop::joy_cb(const sensor_msgs::Joy::ConstPtr &joy_msg){
...
-    ROS_INFO("Init button pressed");
+    ROS_INFO("Init and recover");
+    say("init and recover", true);
 
     for(std::map<std::string,XmlRpc::XmlRpcValue>::iterator p=components_.begin();p!=components_.end();++p)
     {
       std::string comp_name = p->first;
+      say(comp_name, true);
+
+      ROS_INFO("Init %s",comp_name.c_str());
```
Comment (fmessmer) [inline cob_teleop/ros/src/cob_teleop.cpp:404]: Can we have some consistency, i.e. always have `ROS_INFO()` before `say()` without empty line?

**Comment by fmessmer** [comment]
How can we actually fix the "light" issue - without commenting it? Any suggestion?
If we merge this without a fix for "light", it should be documented in #112

File: cob_teleop/ros/src/cob_teleop.cpp (hunk 1 above)
Comment (floweisshardt) [inline cob_teleop/ros/src/cob_teleop.cpp]: addressed in https://github.com/ipa-fmw/cob_command_tools/commit/23cc62115a2c9df18d07c9454f833aff178f5d01

File: cob_teleop/ros/src/cob_teleop.cpp (hunk 2 above)
Comment (floweisshardt) [inline cob_teleop/ros/src/cob_teleop.cpp:404]: addressed in https://github.com/ipa-fmw/cob_command_tools/commit/23cc62115a2c9df18d07c9454f833aff178f5d01

**Comment by floweisshardt** [comment]
I fixed your remarks and added the light issue to #112.

ready for next round...

**Comment by fmessmer** [comment]
would you mind taking care of "tabs vs. spaces" in cob_teleop.cpp and keyboard_publisher.cpp?

**Comment by floweisshardt** [comment]
done in https://github.com/ipa-fmw/cob_command_tools/commit/de8cc45ccad4d044cda4714d6895e4ca824d673d

**Comment by fmessmer** [comment]
Thanks!
Last question: Is jenkins happy with the new cob_teleop or does the build error still exist?

**Comment by floweisshardt** [comment]
still exists, see http://jenkins.ros.org/job/devel-indigo-cob_command_tools/ARCH_PARAM=amd64,UBUNTU_PARAM=trusty,label=devel/37/console

**Comment by fmessmer** [comment]
The jenkins failure is strange....why is including SetLightMode.h possible while including ScriptAction.h fails...where is the difference between `cob_light` and `cob_script_server`?

one thing I can see is that in `cob_light` `actionlib` is defined as `<depend>` and in `cob_script_server` it only is a `<exec_depend>`...

**Comment by floweisshardt** [comment]
merging this and opening issue (#118) for jenkins error

**Pull request merged**